app/backend/hiStoryBackend/api/views/v1/memory_post_views.py

```diff
@@ -1,14 +1,85 @@
-from rest_framework.views import APIView
+import re
+
+from django.contrib.auth.models import AnonymousUser
+from rest_framework import generics, mixins
+from rest_framework.generics import get_object_or_404
+from rest_framework.pagination import PageNumberPagination
+from rest_framework.permissions import IsAuthenticatedOrReadOnly
 
 from api.helpers import json_response_helpers as jrh
+from api.helpers.custom_helpers import IsOwnerOrAdmin
+from api.models import MemoryPost
 from api.serializers import MemoryPostSerializer
 
-class MemoryPostView(APIView):
 
-	def post(self, request, format = None):
-		memory_post_serializer = MemoryPostSerializer(data = request.data)
+class MemoryPostPageNumberPagination(PageNumberPagination):
+	page_size = 10
+
+
+class MemoryPostView(mixins.ListModelMixin,
+					 mixins.RetrieveModelMixin,
+					 mixins.DestroyModelMixin,
+					 generics.GenericAPIView):
+
+	permission_classes = (IsAuthenticatedOrReadOnly, IsOwnerOrAdmin)
+	queryset = MemoryPost.objects.all()
+	serializer_class = MemoryPostSerializer
+	lookup_field = 'id'
+	pagination_class = MemoryPostPageNumberPagination
+
+	def parse_request_data(self, request_data):
+		story_pattern = re.compile(r"^story\[\d+\]$")  # Matches 'story[<digit>]'
+		story_keys = [key for key in request_data.keys() if story_pattern.match(key)]
+		if len(story_keys) == 0:
+			story_arr = None
+		else:
+			story_keys.sort()
+			story_arr = [request_data[key] for key in story_keys]
+
+		data = {'title': request_data.get('title'),
+				'time': request_data.get('time'),
+				'location': request_data.get('location'),
+				'story_arr': story_arr
+				}
+
+		return data
+
+	def get(self, request, *args, **kwargs):
+		if kwargs.get("id"):
+			return self.retrieve(request, *args, **kwargs)  # Comes from RetrieveModelMixin
+		else:
+			queryset = self.get_queryset().order_by('id').reverse()
```

Comment:
Are we going to sort posts by their id? Wouldn't their creation_date or update_date be a better choice?

Reply:
System keeps track of `id`s. When a new model is created, next `id` value is assigned to that model and its `created` field is set automatically. Hence, sorting by `id` means sorting by creation time.